### PR TITLE
omega-h: new package

### PR DESCRIPTION
**Pull request by ibaned**
Adding the Omega_h mesh adaptation library as a package

### CODE REVIEW

File: var/spack/repos/builtin/packages/omega-h/package.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    variant('shared', default=True, description='Build shared libraries')
+    variant('mpi', default=True, description='Activates MPI support')
+    variant('zlib', default=True, description='Activates ZLib support')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('zlib', when='+zlib')
+
+    def cmake_args(self):
+        args = ['-DUSE_XSDK_DEFAULTS:BOOL=ON']
+        libext = 'so'
```
Comment (davydden) [inline var/spack/repos/builtin/packages/omega-h/package.py]: unused, you can remove it.

File: var/spack/repos/builtin/packages/omega-h/package.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        if not '~shared' in self.spec:
+            args.append('-DBUILD_SHARED_LIBS:BOOL=ON')
+        else:
+            args.append('-DBUILD_SHARED_LIBS:BOOL=OFF')
+            libext = 'a'
```
Comment (davydden) [inline var/spack/repos/builtin/packages/omega-h/package.py]: same

File: var/spack/repos/builtin/packages/omega-h/package.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        if not '~mpi' in self.spec:
```
Comment (davydden) [inline var/spack/repos/builtin/packages/omega-h/package.py]: `if '+mpi' in ...`

same elsewhere.

**Comment by ibaned** [comment]
@davydden thanks for the quick feedback! Fixing these now...

**Comment by ibaned** [comment]
I'm not sure if I can do anything about the `codecov/project` failure? Please let me know if I can.

**Pull request merged**